src/controllers/dimension-processor.ts

```diff
@@ -0,0 +1,137 @@
+                        throw new Error('Only one DataValues source can be specified');
+                    }
+                    source.action = SourceAction.CREATE;
+                    datavalues = sourceInfo;
+                    break;
+                case SourceType.FOOTNOTES:
+                    if (footnotes) {
+                        throw new Error('Only one FootNote source can be specified');
+                    }
+                    source.action = SourceAction.CREATE;
+                    footnotes = sourceInfo;
+                    break;
+                case SourceType.DIMENSION:
+                    source.action = SourceAction.CREATE;
+                    dimensions.push(sourceInfo);
+                    break;
+                case SourceType.IGNORE:
+                    source.action = SourceAction.IGNORE;
+                    ignore.push(sourceInfo);
+                    break;
+                default:
+                    throw new Error(`Invalid source type: ${sourceInfo.sourceType}`);
+            }
+            source.type = sourceInfo.sourceType;
+            await source.save();
+        })
+    );
+    return { datavalues, footnotes, dimensions, ignore };
+};
+
+export const createDimensions = async (
+    revision: Revision,
+    validatedDimensionCreationRequest: ValidatedDimensionCreationRequest
+): Promise<Dataset> => {
+    const dataset = await revision.dataset;
+    if (!dataset) {
+        throw new Error('No dataset is attached to this revision');
+    }
+    const languages = i18next.languages;
+    const { footnotes, dimensions } = validatedDimensionCreationRequest;
+    if (footnotes) {
+        const footnoteDimension = new Dimension();
+        footnoteDimension.id = crypto.randomUUID().toLowerCase();
+        const footnoteDimensionInfo: DimensionInfo[] = [];
+        const updateDate = new Date(Date.now());
+        footnoteDimension.dimensionInfo = Promise.resolve(
+            languages.map((lang) => {
+                const dimensionInfo = new DimensionInfo();
+                dimensionInfo.dimension = Promise.resolve(footnoteDimension);
+                dimensionInfo.language = lang;
+                dimensionInfo.name = t('dimension_info.footnotes.name', { lng: lang });
+                dimensionInfo.description = t('dimension_info.footnotes.description', { lng: lang });
+                dimensionInfo.updatedAt = updateDate;
+                return dimensionInfo;
+            })
+        );
+        footnoteDimension.dimensionInfo = Promise.resolve(footnoteDimensionInfo);
+        footnoteDimension.type = DimensionType.FOOTNOTE;
+        footnoteDimension.dataset = Promise.resolve(dataset);
+        footnoteDimension.startRevision = Promise.resolve(revision);
+        const source = await Source.findOne({ where: { id: footnotes.sourceId } });
+        if (!source) {
+            throw new Error(`Source with id ${footnotes.sourceId} not found`);
+        }
```

Comment:
again... This resolves the issue in typescript that the result of a findOne can result in an undefined so to prevent a compiler issue you have to check to make sure the thing you requested really is a thing rather than an undefined.

Reply:
Try replacing `findOne(...)` with `findOneOrFail(...)`